### PR TITLE
Lower complexity @ Form:isValid

### PR DESCRIPTION
**Pull request by krzysiekpiasecki**
| Q | A |
| --- | --- |
| Branch? | 3.0 for fixes |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | n/a |
| License | MIT |
| Doc PR | n/a |

Why?
- lower LOC
- less return expressions
- lower complexity
- more readable comparison

### CODE REVIEW

File: src/Symfony/Component/Form/Form.php [hunk 1]
```diff
@@ -731,11 +731,7 @@ public function isValid()
             return true;
         }
 
-        if (count($this->getErrors(true)) > 0) {
-            return false;
-        }
-
-        return true;
+        return count($this->getErrors(true)) === 0;
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Form/Form.php]: could even be reduced to `return !$this->getErrors(true);`

Comment (nicolas-grekas) [inline src/Symfony/Component/Form/Form.php]: nevermind, getErrors return a coutable...

**Comment by nicolas-grekas** [comment]
:+1:

**Comment by derrabus** [comment]
This patch should rather be applied to the 2.7 branch.

File: src/Symfony/Component/Form/Form.php (hunk 1 above)
Comment (hhamon) [inline src/Symfony/Component/Form/Form.php]: It should be:

```php
return 0 === count($this->getErrors(true));
```

**Comment by krzysiekpiasecki** [comment]
Replace with #18400

**Pull request closed without merging**